nexus/reconfigurator/planning/src/mgs_updates/mod.rs

```diff
@@ -476,6 +652,179 @@ fn try_make_update_sp(
     })
 }
 
+/// Determine if the given baseboard needs an SP update and, if so, returns it.
+fn try_make_update_rot(
+    log: &slog::Logger,
+    baseboard_id: &Arc<BaseboardId>,
+    inventory: &Collection,
+    current_artifacts: &TufRepoDescription,
+) -> Option<PendingMgsUpdate> {
+    let Some(sp_info) = inventory.sps.get(baseboard_id) else {
+        warn!(
+            log,
+            "cannot configure RoT update for board \
+             (missing SP info from inventory)";
+            baseboard_id
+        );
+        return None;
+    };
+
+    let Some(rot_state) = inventory.rots.get(baseboard_id) else {
+        warn!(
+            log,
+            "cannot configure RoT update for board \
+             (missing RoT state from inventory)";
+            baseboard_id
+        );
+        return None;
+    };
+
+    let active_slot = rot_state.active_slot;
+
+    let active_caboose = match active_slot {
+        RotSlot::A => CabooseWhich::RotSlotA,
+        RotSlot::B => CabooseWhich::RotSlotB,
+    };
+
+    let Some(active_caboose) =
+        inventory.caboose_for(active_caboose, baseboard_id)
+    else {
+        warn!(
+            log,
+            "cannot configure RoT update for board \
+             (missing active slot {active_slot} caboose from inventory)";
+            baseboard_id,
+        );
+        return None;
+    };
+
+    let Ok(expected_active_version) = active_caboose.caboose.version.parse()
+    else {
+        warn!(
+            log,
+            "cannot configure RoT update for board \
+             (cannot parse current active version as an ArtifactVersion)";
+            baseboard_id,
+            "found_version" => &active_caboose.caboose.version,
+        );
+        return None;
+    };
+
+    let board = &active_caboose.caboose.board;
+    let matching_artifacts: Vec<_> = current_artifacts
+        .artifacts
+        .iter()
+        .filter(|a| {
+            // A matching RoT artifact will have:
+            //
+            // - "name" matching the board name (found above from caboose)
+            // - "kind" matching one of the known RoT kinds
```

Comment:
For the RoT, don't we need to check that it's signed with the same key (`SIGN` in caboose matches something in the metadata)?

I think the corresponding Wicket code path uses a common `verify` function to do this (sorry, I don't have the reference handy at the moment).  I wonder if we need to be using the same thing.  (I wouldn't say we necessarily do.  We already duplicate some of the Wicket behavior because the shape of the code is different.  But we should at least make sure it covers the same logic, and if it's non-trivial, think about commonizing it.)

Reply:
> To do this I'll need to make API calls directly to MGS, this isn't done in the planner yet. I'm assuming it's OK to do so.

I think we should strongly prefer not doing so; the planner currently operates as essentially a pure function of its inputs, and it'd be very nice to not lost that. I think in this case, the latest inventory may already have what you need: inventory collections [include CMPA/CFPA pages](https://github.com/oxidecomputer/omicron/blob/b7b24d5f1eefb989cadf97f662fdc8235130ac26/nexus/types/src/inventory.rs#L114-L122). If we need more than that from MGS, can we add it to inventory instead of making the planner ask directly?

Reply:
We chatted about this in the update watercooler. We don't want to pass the entire artifacts to the planner and came up with a couple of possible paths forward:

1. The planner could give a list of all the possible artifacts to the executor, and the executor verifies the artifacts the same way wicket does today.
2. We somehow extract this information from an artifact _before_, and include it as part of the TUF metadata.

I'll catch up with @plotnick next week to take a swing at option #2 which is much nicer.